src/main/java/core/basesyntax/db/Storage.java

```
@@ -0,0 +1,21 @@
+package core.basesyntax.db;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private static final Map<String, Integer> fruitsStorage = new HashMap<>();
+
+    public static void putFruitQuantity(String fruit, int quantity) {
+        fruitsStorage.put(fruit, quantity);
+    }
+
+    public static int getFruitQuantity(String fruit) {
+        return fruitsStorage.getOrDefault(fruit, 0);
```

getFruitQuantity accepts any `fruit` value and returns 0 for missing or possibly for a null key call. The requirements require validating inputs: "Validate inputs to storage operations (non-null fruit, non-negative quantities)." You should check that `fruit` is non-null and non-blank and throw an informative RuntimeException rather than silently treating null as missing.